protocol/packages/currencies/src/payment/mod.rs

```diff
@@ -1,45 +1,51 @@
-use currency::{AnyVisitor, Group, Matcher, MaybeAnyVisitResult, MemberOf};
-use sdk::schemars::{self, JsonSchema};
 use serde::{Deserialize, Serialize};
 
-use super::{lease::LeaseGroup, lpn::Lpns, native::Native};
+use currency::{AnyVisitor, Matcher, MaybeAnyVisitResult, MemberOf};
+use sdk::schemars::JsonSchema;
 
-pub use self::only::PaymentOnlyGroup;
+use crate::{lease, lpn, native};
 
+pub use self::only::Group as OnlyGroup;
 #[cfg(feature = "testing")]
-pub use testing::*;
+pub use self::testing::*;
 
 pub(crate) mod only;
 #[cfg(feature = "testing")]
 mod testing;
 
 #[derive(
-    Clone, Copy, Debug, Ord, PartialEq, PartialOrd, Eq, JsonSchema, Serialize, Deserialize,
+    Clone, Copy, Debug, Ord, PartialEq, PartialOrd, Eq, Serialize, Deserialize, JsonSchema,
 )]
-pub struct PaymentGroup {}
+#[serde(deny_unknown_fields, rename_all = "snake_case")]
+#[schemars(crate = "sdk::schemars")]
+pub enum Group {}
 
-impl Group for PaymentGroup {
+impl currency::Group for Group {
     const DESCR: &'static str = "payment";
+
     type TopG = Self;
 
+    #[inline]
     fn maybe_visit<M, V>(matcher: &M, visitor: V) -> MaybeAnyVisitResult<Self, V>
     where
         M: Matcher,
         V: AnyVisitor<Self>,
     {
-        LeaseGroup::maybe_visit_member(matcher, visitor)
-            .or_else(|visitor| Lpns::maybe_visit_member(matcher, visitor))
-            .or_else(|visitor| Native::maybe_visit_member(matcher, visitor))
-            .or_else(|visitor| PaymentOnlyGroup::maybe_visit_member(matcher, visitor))
+        lease::Group::maybe_visit_member(matcher, visitor)
+            .or_else(|visitor| lpn::Group::maybe_visit_member(matcher, visitor))
+            .or_else(|visitor| native::Group::maybe_visit_member(matcher, visitor))
+            .or_else(|visitor| only::Group::maybe_visit_member(matcher, visitor))
     }
 
-    fn maybe_visit_member<M, V>(_matcher: &M, _visitor: V) -> MaybeAnyVisitResult<Self::TopG, V>
+    #[cold]
+    #[inline]
+    fn maybe_visit_member<M, V>(_: &M, _: V) -> MaybeAnyVisitResult<Self::TopG, V>
     where
         M: Matcher,
         V: AnyVisitor<Self::TopG>,
     {
-        unreachable!()
+        unimplemented!()
```

Comment:
disagree
the meaning of `unreachable` better match the real case, please revert

Reply:
I changed it because the documentation of the `unreachable!` macro states that:

> This is useful any time that the compiler can’t determine that some code is unreachable. For example:
>
> * Match arms with guard conditions.
> * Loops that dynamically terminate.
> * Iterators that dynamically terminate.

I'd argue that our case does not fit any of these and it's more of an API design issue, thus `unimplemented!` fits better in my personal opinion.

Reply:
I would even go as far to suggest splitting the two traits if we even need to put a marker that this method is not implemented and/or shouldn't be executed.

Reply:
The code is prose. The `unimplemened!` conveys the developer's intent to implement it at some point of time. In our case, that is not true.
The best solution would be to transform the trait(s) to avoid such cases, but since it requires additional work I insist on reverting to the main branch state.

Reply:
I disagree with the first statement because this is from the documentation of `todo!`:
> The difference between `unimplemented!` and `todo!` is that while `todo!` conveys an intent of implementing the functionality later and the message is “not yet implemented”, `unimplemented!` makes no such claims. Its message is “not implemented”.

For the second, if you agree, could you write it down as an issue so we can pick it up at a later date?

Reply:
After digging more into their descriptions, I have to admit that I have been using `unreachable` in cases it was not intended to be.  Since there are more occurrences in the code, I'd suggest changing that in a dedicated commit/PR. 
https://github.com/nolus-protocol/nolus-money-market/issues/445

Reply:
Okay. That sounds good to me! Reverting the change here now.